src/test/app/Vault_test.cpp

```diff
@@ -2843,6 +3190,870 @@ class Vault_test : public beast::unit_test::suite
         env(tx, ter{terADDRESS_COLLISION});
     }
 
+    void
+    testScaleIOU()
+    {
+        using namespace test::jtx;
+
+        struct Data
+        {
+            Account const& owner;
+            Account const& issuer;
+            Account const& depositor;
+            Account const& vaultAccount;
+            MPTIssue shares;
+            PrettyAsset const& share;
+            Vault& vault;
+            ripple::Keylet keylet;
+            Issue assets;
+            PrettyAsset const& asset;
+            std::function<bool(std::function<bool(SLE&, SLE&)>)> peek;
+        };
+
+        auto testCase = [&, this](
+                            std::uint8_t scale,
+                            std::function<void(Env & env, Data data)> test) {
+            Env env{*this, testable_amendments() | featureSingleAssetVault};
+            Account const owner{"owner"};
+            Account const issuer{"issuer"};
+            Account const depositor{"depositor"};
+            Vault vault{env};
+            env.fund(XRP(1000), issuer, owner, depositor);
+            env(fset(issuer, asfAllowTrustLineClawback));
+            env.close();
+
+            PrettyAsset const asset = issuer["IOU"];
+            env.trust(asset(1000), owner);
+            env.trust(asset(1000), depositor);
+            env(pay(issuer, owner, asset(200)));
+            env(pay(issuer, depositor, asset(200)));
+            env.close();
+
+            auto [tx, keylet] = vault.create({.owner = owner, .asset = asset});
+            tx[sfScale] = scale;
+            env(tx);
+
+            auto const [vaultAccount, issuanceId] =
+                [&env](ripple::Keylet keylet) -> std::tuple<Account, MPTID> {
+                auto const vault = env.le(keylet);
+                return {
+                    Account("vault", vault->at(sfAccount)),
+                    vault->at(sfShareMPTID)};
+            }(keylet);
```

Comment:
No objection, but is there a reason you pass `keylet` as a parameter instead of capturing it?

Reply:
I do not remember which compiler version was that, but we had problems with capture of variables introduced from tuple-like structured binding declaration, so `keylet` had to be captured (for some compiler) using explicitly capture variable. You can see this applied through this file; I decided to use simpler syntax in this case instead.

Reply:
> I do not remember which compiler/version was that, but we had problems with capture of variables introduced from tuple-like structured binding declaration, so `keylet` had to be captured (for some compiler) using explicitly captured variable. You can see this applied through this file; I decided to use simpler syntax in this case instead.

That makes sense!